### ISSUE TITLE
Question about the loss function

### ISSUE THREAD

**Issue by ghost** [ushasi/A-Simplified-framework-for-Zero-shot-Cross-Modal-Sketch-Data-Retrieval#4]
Dear author, thank you for your excellent job!
But when I read your paper and code, I have encountered some problems.

1. The loss function in your code is not consistent with your paper:
Unified_XY_triads.py, line 411, 
`cost = 0.1*loss0 + 0*loss3 + 0.1*loss4 + 1*loss5+ loss6 + loss7 #+loss8 #  + 0*loss9     # MODEL 2`
doesn't have all loss as your paper claims.
2. In your paper, Section 3.3, 

> Since we have multiple losses, our problem becomes a non-convex optimization problem. To solve this, we reduce each loss term individually, keeping the other losses constant. The problem transforms into a convex optimization problem for that loss. 

I don't understand. Neural Network is a non-convex function itself.  Is a possible to change it to convex function by changing the loss function?

**Comment by ghost** [comment]
BTW, why you use GCN?
`graph = sg.StellarGraph(nodes, edges)`， line129
